src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -1124,6 +1128,20 @@ public synchronized SharedWorkerPool createSharedWorkerPool(String name, int poo
     return sharedWorkerPool;
   }
 
+  private ThreadFactory createThreadFactory(long maxExecuteTime, TimeUnit maxExecuteTimeUnit, String prefix, boolean worker) {
+    AtomicInteger threadCount = new AtomicInteger(0);
+    return runnable -> {
+      VertxThread thread = threadFactory.newVertxThread(runnable, prefix + threadCount.getAndIncrement(), worker, maxExecuteTime, maxExecuteTimeUnit);
+      checker.registerThread(thread, thread);
+      // Vert.x threads are NOT daemons - we want them to prevent JVM exit so embededd user doesn't
+      // have to explicitly prevent JVM from exiting.
+      // I know the default is false anyway, but just to be explicit-  Vert.x threads are NOT daemons
+      // we want to prevent the JVM from exiting until Vert.x instances are closed
```

Comment:
Seems like the comment is duplicated :)

Reply:
Also, should this really a lambda?
I don't know about Vert.x, but in Quarkus we try to stay away from lambdas as they have a slightly higher metaspace cost than regular classes.

Reply:
it's a lambda for readability, we have many lambda in other places and it seems it never raised an issue.